### ISSUE TITLE
Code Review 2

### ISSUE THREAD

**Issue by maldon01** [shnaky/AdversarialAttacks.jl#101]
I had no issues downloading the package or running the example, very good! 

## Documentation:

- A small paragraph at the start of the documentation including the motivation and what the package is about would be nice. 
- **Structure**: overall good. Since there is a page named black box attacks I would recommend changing the name of the page fast gradient sign method to white box attack for more consistency and bringing the user interface to the top, so it is more intuitive when navigating the documentation for the first time. 
- There is some methods that are found twice in the documentation (both under AdversarialAttack and the corresponding attack type). 
- Just from the documentation is not so clear what models can be used with each attack type or in general with the package. How should custom models be handled? 
- **Example**: very nice to include the evaluation in the getting started. However I found some visualization was missing. During CR2 I was shown some visuals to the project, so I could see both the original image, the changes performed during the attack and the final result. Together with the prediction accuracy of the model before and after the attack that was enough to understand what the package was doing. I think it would be very helpful for the user to provide some examples with visualization of the images (like the ones included in the examples folder of the repository) in the getting started 

## Code: 
- Good folder structure, only slightly confusing to have the craft methods in the attacks folder and attack methods in Interface. (Also is it really needed to export craft? In the provided examples only attack was used). 
- Missing import needed for: https://github.com/shnaky/AdversarialAttacks.jl/blob/484700057ba169b115529e638d74aa1a052d4736/src/attacks/BlackBoxAttacks.jl#L110. In general better write imports in main and in an explicit manner. 
- **Generalization**
1. I like the idea of making the package work for a variety of models with the AbstractModel abstraction. However it still seems not 100% implemented since the decision tree model needs special treatment. https://github.com/shnaky/AdversarialAttacks.jl/blob/484700057ba169b115529e638d74aa1a052d4736/src/attacks/BlackBoxAttacks.jl#L136 
2. As far as I know models can return logits and probabilities, for good generalization with user defined models (in combination with the wrapper) make sure you separate these two carefully for FGSM where you need to calculate the gradients. 
- **Evaluation**: I am not an expert in the topic, but I think it would be maybe more intuitive to not include all samples (including misclassified ones) in the metric of the adversarial attack, so you just compute how many of the correctly classified samples are affected by the attack instead of overall accuracy. Just a suggestion. https://github.com/shnaky/AdversarialAttacks.jl/blob/484700057ba169b115529e638d74aa1a052d4736/src/Evaluation.jl#L154 

## Test
Good test coverage score
